### PR TITLE
PS-256: keys created by keyring_vault MTR tests are not removed from

### PR DESCRIPTION
**Pull request by robgolebiowski**
Vault server

This fix changes how keys are removed from Vault. Instead of removing
one key after another, test:
 1) creates a mount point before running the tests.
 2) run actual test (which creates/removes keys in/from mount point
    created in step 1
 3) removes the mount point created in step 1

### CODE REVIEW

**Comment by robgolebiowski** [comment]
http://jenkins.percona.com/view/5.7/job/mysql-5.7-param/1311/

File: plugin/keyring_vault/tests/mtr/mount_point_service.inc [hunk 1]
```diff
@@ -0,0 +1,62 @@
+# Creates or deletes secret mount point specified in keyring_vault configuration file
+# Mount point is :
+#  - created when MOUNT_POINT_SERVICE_OP is set to 1
```
Comment (laurynas-biveinis) [inline plugin/keyring_vault/tests/mtr/mount_point_service.inc]: Why not use "CREATE", "DELETE" or some other descriptive strings instead of 1 and 2?

**Review by laurynas-biveinis — CHANGES_REQUESTED** [review]

File: plugin/keyring_vault/tests/mtr/mount_point_service.inc (hunk 1 above)
Comment (robgolebiowski) [reply plugin/keyring_vault/tests/mtr/mount_point_service.inc]: The problem is where to define those CREATE and DELETE. Test includes mount_point_service.inc and it needs to define what operation it wants mount_point_service to perform. The perfect solution would be --let MOUNT_POINT_SERVICE_OP=CREATE and then source mount_point_service.inc, but the CREATE would need to be defined before this source operation and in every test that source mount_point_service.inc

Comment (laurynas-biveinis) [reply plugin/keyring_vault/tests/mtr/mount_point_service.inc]: Not CREATE but "CREATE" ?

Comment (robgolebiowski) [reply plugin/keyring_vault/tests/mtr/mount_point_service.inc]: ahh, right, changing

File: plugin/keyring_vault/tests/mtr/mount_point_service.inc [hunk 2]
```diff
@@ -0,0 +1,62 @@
...
+    {
+      $secret_mount_point=$1;
+    }
+    elsif ($row =~ m/vault_ca[ ]*= [ ]*(.*)/)
+    {
+      $vault_ca=$1;
+    }
+  }
+  close(CONF_FILE);
+  if ($token eq "" || $vault_url eq "" || $secret_mount_point eq "")
```
Comment (percona-ysorokin) [inline plugin/keyring_vault/tests/mtr/mount_point_service.inc:45]: `|| $vault_ca eq ""`? Or is it optional?

Comment (robgolebiowski) [reply plugin/keyring_vault/tests/mtr/mount_point_service.inc:45]: it's optional

Comment (percona-ysorokin) [reply plugin/keyring_vault/tests/mtr/mount_point_service.inc:45]: Then you need to take into account this when you form `curl` command lines for both `CREATE` and `DELETE`
```perl
if($vault_ca)
{
    system(qq#curl -H "X-Vault-Token: $token" --cacert $vault_ca -X DELETE $vault_url/v1/sys/mounts/$secret_mount_point#);
}
else
{
    system(qq#curl -H "X-Vault-Token: $token" -X DELETE $vault_url/v1/sys/mounts/$secret_mount_point#);
}
```

File: plugin/keyring_vault/tests/mtr/mount_point_service.inc (hunk 1 above)
Comment (robgolebiowski) [reply plugin/keyring_vault/tests/mtr/mount_point_service.inc]: Done

File: plugin/keyring_vault/tests/mtr/mount_point_service.inc (hunk 2 above)
Comment (robgolebiowski) [reply plugin/keyring_vault/tests/mtr/mount_point_service.inc:45]: Done

**Comment by robgolebiowski** [comment]
JR => https://jenkins.percona.com/view/5.7/job/mysql-5.7-param/1393/

**Review by percona-ysorokin — APPROVED** [review]
LGTM

File: plugin/keyring_vault/tests/mtr/key_rotation_qa.test [hunk 3]
```diff
@@ -27,6 +27,13 @@ call mtr.add_suppression("\\[ERROR\\] InnoDB: Failed to decrpt encryption inform
 
 --source generate_default_conf_files.inc
 
+# Create mount points
+--let MOUNT_POINT_SERVICE_OP=CREATE
```
Comment (laurynas-biveinis) [inline plugin/keyring_vault/tests/mtr/key_rotation_qa.test:31]: let FOO= vs let $BAR=
inconsistency on the dollar sign. By itself too minor to force a PR respin

**Review by laurynas-biveinis — APPROVED** [review]

**Comment by laurynas-biveinis** [comment]
This has changed RocksDB submodule pointer while it is not RocksDB-related, probably broke the its builds

**Comment by laurynas-biveinis** [comment]
https://bugs.launchpad.net/percona-server/+bug/1740350

**Pull request merged**